rustls/src/client/ech.rs

```diff
@@ -292,6 +312,54 @@ impl EchState {
         Ok(outer_hello)
     }
 
+    /// Confirm whether an ECH offer was accepted based on examining the server hello.
+    pub(crate) fn confirm_acceptance(
+        self,
+        ks: &mut KeyScheduleHandshakeStart,
+        server_hello: &ServerHelloPayload,
+        hash: &'static dyn Hash,
+    ) -> Result<Option<EchAccepted>, Error> {
+        // Start the inner transcript hash now that we know the hash algorithm to use.
+        let inner_transcript = self
+            .inner_hello_transcript
+            .start_hash(hash);
+
+        // Fork the transcript that we've started with the inner hello to use for a confirmation step.
+        // We need to preserve the original inner_transcript to use if this confirmation succeeds.
+        let mut confirmation_transcript = inner_transcript.clone();
+
+        // Add the server hello confirmation - this differs from the standard server hello encoding.
+        confirmation_transcript.add_message(&Self::server_hello_conf(server_hello));
+
+        // Derive a confirmation secret from the inner hello random and the confirmation transcript.
+        let derived = ks.server_ech_confirmation_secret(
+            self.inner_hello_random.0.as_ref(),
+            confirmation_transcript.current_hash(),
+        );
+
+        // Check that first 8 digits of the derived secret match the last 8 digits of the original
+        // server random. This match signals that the server accepted the ECH offer.
+        // Indexing safety: Random is [0; 32] by construction.
+        Ok(
```

Comment:
Style nit: I think I've come to the conclusion that hoisting the `Ok` might not be worth it if it causes 2 extra lines + an extra level of indentation.

Reply:
Fixed.